lib/active_resource/base.rb

```diff
@@ -1620,19 +1620,25 @@ def find_or_create_resource_for(name)
         resource_name = name.to_s.camelize
 
         const_args = [resource_name, false]
-        if self.class.const_defined?(*const_args)
-          self.class.const_get(*const_args)
-        else
-          ancestors = self.class.name.to_s.split("::")
-          if ancestors.size > 1
-            find_or_create_resource_in_modules(resource_name, ancestors)
+
+        begin
+          if self.class.const_defined?(*const_args)
+            self.class.const_get(*const_args)
           else
-            if Object.const_defined?(*const_args)
-              Object.const_get(*const_args)
+            ancestors = self.class.name.to_s.split("::")
+            if ancestors.size > 1
+              find_or_create_resource_in_modules(resource_name, ancestors)
             else
-              create_resource_for(resource_name)
+              if Object.const_defined?(*const_args)
+                Object.const_get(*const_args)
+              else
+                create_resource_for(resource_name)
+              end
             end
           end
+        rescue NameError
```

Comment:
Narrowing from StandardError to NameError helps, but we're still capturing exceptions that may be raised for other reasons. We'd like to only capture NameError raised when we call `const_defined?`, so let's only wrap that one call if we can. Possible route:
```ruby
if const_valid?(*const_args)
  …
else
  find_or_create_resource_for(:UnknownResource)
end

def const_valid?(*const_args)
  self.class.const_defined?(*const_args)
  true
rescue NameError
  false
end
```

Reply:
Oh I like that better too. I'll change that now